plugins/pulp_rpm/plugins/importers/yum/depsolve.py

```diff
@@ -421,8 +432,6 @@ def get_requirements(self, units):
         :rtype:     generator
         """
         for segment in paginate(units):
-            unit_ids = [unit.id for unit in segment]
-            fields = ['requires', 'id']
-            for result in models.RPM.objects.filter(id__in=unit_ids).only(*fields):
-                for require in result.requires or []:
+            for unit in segment:
+                for require in unit.requires or []:
```

Comment:
Hokay.

I took a closer look at this. Here are the bullet points:
- `units` comes from outside the module, and the docblock was definitely wrong. It's an interable of mongoengine RPM unit instances. The units in the incoming units iterable do not need to be converted to units.
- This method consumes units and yields Requirements. There is nothing downstream that will load these units into RAM, unless it's done with a copy of the units iterable that was generated outside of this module, so that is outside of the concerns of this module.
- This change is already doing conventional things to avoid the wrath of oom-killer, so if anything I think that this change reduces the number of DB queries, RAM usage, and (as a result) risk.
- My reasons for why it may not be totally weird to re-query the database, upon closer inspection, were based on incorrect assumptions that I made, and I've discarded them.
- I discussed these points just now with rbarlow, and the LGTM was sustained. :)